src/javascript/binary/pages/portfiliows.js

```diff
@@ -0,0 +1,209 @@
+            currency = c.currency;
+            contracts += rowTemplate
+            .split("!transaction_id!").join(c.transaction_id)
+            .split("!contract_id!").join(c.contract_id)
+            .split("!longcode!").join(c.longcode)
+            .split("!currency!").join(c.currency)
+            .split("!buy_price!").join(fixCurrency(c.buy_price));
+        });
+
+        // contracts is ready to be added to the dom
+        $("#portfolio-dynamic").replaceWith(trans(contracts));
+
+        // update footer area data
+        sumPurchase = sumPurchase.toFixed(2);
+        $("#cost-of-open-positions").text( fixCurrency(sumPurchase, currency));
+
+        // request "proposal_open_contract"
+        BinarySocket.send({"proposal_open_contract":1});
+
+        // ready to show portfolio table
+        $("#trading_init_progress").remove();
+        $("#portfolio-content").removeClass("dynamic");
+
+    };
+
+    var updateIndicative = function(data) {
+
+        if(data.proposal_open_contract.is_valid_to_sell != 1) {
+            $("tr[data-contract_id='"+data.proposal_open_contract.contract_id+"'] td.indicative").text(text.localize('Resale not offered'));
+            return false;
+        }
+
+        $("tr[data-contract_id='"+data.proposal_open_contract.contract_id+"'] td.indicative").html(data.proposal_open_contract.currency+' <strong class="indicative_price">'+data.proposal_open_contract.bid_price+'</strong>');
+
+        var indicative_sum = 0, indicative_price = 0;
+        $("strong.indicative_price").each(function() {
+            indicative_price = $(this).text();
+            indicative_price = parseFloat(indicative_price, 2);
+            if(!isNaN(indicative_price)) {
+                indicative_sum += indicative_price;
+            }
+        });
+
+        indicative_sum = indicative_sum.toFixed(2);
+
+        $("#value-of-open-positions").text(fixCurrency(indicative_sum, "USD"));
+
+    };
+
+
+    /*** utility functions ***/
+
+    // Dynamic text
+    var dTexts = ["view", "indicative"];
+
+    /**
+     * In the dynamic parts we have strings to include
+     * For instance, in portfolio table, we have a 'View' button
+     * for each contract.
+    **/
+    var trans = function(str) {
+        var placeholder;
+        for(var i = 0, l = dTexts.length; i < l; i++) {
+            placeholder = ":"+dTexts[i]+":";
+            if(-1 === str.indexOf(placeholder)) continue;
+            str = str.split(placeholder).join(text.localize(dTexts[i]));
+        }
+        return str;
+    };
+
+    /**
+     * Amounts received from the API could be integer or decimal numbers.
+     * In case we have an integer, like 73, we want to display a decimal
+     * with two fractions, i.e. 73:00
+     * This function does that.
+     * Adapted from http://stackoverflow.com/a/14428340
+     * Kudos to: [VisioN](http://stackoverflow.com/users/1249581)
+    **/
+    var fixCurrency = function(n, c) {
```

Comment:
this is a common utility function. Can you move it to src/javascript/binary/static/common/string_util.js ?